src/DependencyInjection/Configuration.php

```diff
@@ -339,6 +342,133 @@ private function addConnectionsSection(ArrayNodeDefinition $rootNode): void
                                     ->end()
                                 ->end()
                             ->end()
+                            ->arrayNode('autoEncryption')
+                                ->children()
+                                    ->booleanNode('bypassAutoEncryption')->end()
+                                    ->scalarNode('keyVaultClient')->end()
+                                    ->scalarNode('keyVaultNamespace')
+                                        ->validate()
+                                            ->ifTrue(static fn ($v) => ! preg_match('/^.+\..+$/', $v))
+                                            ->thenInvalid('Invalid keyVaultNamespace format. It should be "database.collection".')
+                                        ->end()
+                                    ->end()
+                                    ->arrayNode('masterKey')
```

Comment:
Why is the `masterKey` defined under `autoEncryption` here and in XML?

Do I understand correctly that this is only used for `Database::createEncryptedCollection()`? In that case I wonder if it's something that should instead be configured at the model level (per collection), with this possibly renamed to `defaultMasterKey`.